content/Hardware Support/UNO/Update-the-connectivity-module-firmware-on-UNO-R4-WiFi.md

```diff
@@ -204,3 +259,71 @@ If you get a `VCRUNTIME140.dII was not found` error, download and install the <a
 1. Close the terminal window.
 1. Disconnect the USB cable of the **UNO R4 WiFi** board from your computer.
 1. Connect the **UNO R4 WiFi** board to your computer with the USB cable again.
+
+---
+
+<a id="espflash"></a>
+
+## Run espflash directly
+
+The [updater script](#unor4wifi-updater) will not work if the board cannot be identified as a UNO R4 WiFi. This can happen if the custom firmware for the ESP32-S3 connectivity module is missing completely, or is not functioning correctly. However, the board can still be restored by [running espflash directly](https://github.com/arduino/uno-r4-wifi-usb-bridge/tree/main/unor4wifi-updater#option-2).
```

Comment:
```suggestion
The [Firmware Updater in Arduino IDE](#ide) and the [updater script](#unor4wifi-updater) will not work if the board cannot be identified as a UNO R4 WiFi. This can happen if the custom firmware for the ESP32-S3 connectivity module is missing completely, or is not functioning correctly. However, the board can still be restored by [running espflash directly](https://github.com/arduino/uno-r4-wifi-usb-bridge/tree/main/unor4wifi-updater#option-2).
```
The [Firmware Updater in Arduino IDE](#ide) will also not work if the board is not identified as UNO R4 (as written in the note in the [Use the Firmware Updater in Arduino IDE](#ide) section)

Reply:
That's a good point, but we might as well expand it to also cover the IoT Cloud method:

```suggestion
The above methods will not work if the board cannot be identified as a UNO R4 WiFi. This can happen if the custom firmware for the ESP32-S3 connectivity module is missing completely, or is not functioning correctly. However, the board can still be restored by [running espflash directly](https://github.com/arduino/uno-r4-wifi-usb-bridge/tree/main/unor4wifi-updater#option-2).
```